maint/extractcvars.in

```diff
@@ -26,6 +26,7 @@ $Text::Wrap::unexpand = 0; # disable hard tabs in output
 # set true to enable debug output
 my $debug = 0;
 my @cvars = ();
+my %cvars; # used for duplication detection
```

Comment:
naming both these data structures with the same name gives me pause. i'm surprised its even allowed.

Reply:
This is what makes Perl great -- the sigils. Both variables are for the same semantics, thus having different names is more confusing to read than the same name, in my view. The sigils denote syntactic meanings. It is perfectly okay because Arrays and Hashes use different namespaces. They are different variables from Perl's perspective.

Reply:
To me, the difference is too subtle. I had to look up to understand why it even worked. I think it is bad language feature. I'd rather the variables be cvars_array and cvars_hash, but I'm not sure I care enough to dig in.

Reply:
I can change them to `cvar_array` and `cvar_hash`. It is mostly the language familiarity issue though. The sigils are always part of the names to me, like the `_array`, `_hash` suffix.

Reply:
No need to change it. Another reason to dislike perl, though 😅.